### PR TITLE
Add specs to confirmTransaction

### PR DESCRIPTION
**Pull request by kraytruthg**
Add specs to confirmTransaction

The following pending specs were removed:
- it("sets mediation expiry to 0 when mediator raises an error")
- it("passes the transaction's amount to the mediator")
- it("transfers the mediator fee to the mediator")
- it("collects 0 fee when mediator raises an error")
- it("collects 0 fee when mediator returns a fee higher than the transaction amount")

### CODE REVIEW

**Comment by kraytruthg** [comment]
The state `ConfirmedAfterEscalation` is not handled in `_fetchMediatorFee`.
So specs for **Escalation** are pending.

**Comment by kraytruthg** [comment]
@ngan 

The mediator fee for these events should always be **0**, right?
Why do we choose to record it in the event?
``` solidity
  // Event emitted when a transaction is confirmed by the buyer after the
  // mediation expiry.
  event TransactionConfirmedAfterEscalation(
    uint256 indexed id,
    uint256 mediatorFee
  );

  // Event emitted when a transaction is refunded by the seller after the
  // mediation expiry.
  event TransactionRefundedAfterEscalation(
    uint256 indexed id,
    uint256 mediatorFee
  );
```

File: test/confirmTransaction.js [hunk 1]
```diff
@@ -2,48 +2,352 @@ const $util = require("./util")
...
-    it("fails for mediator")
-    it("fails for policy")
-    it("fails for unknown address")
-    it("fails when transaction does not exist")
+    it("fails for seller", async () => {
+      let {
+        protocol,
+        transaction
+      } = await $util.buildTransaction(
+        buyer, seller, { finalState: $util.states.Accepted }
```
Comment (ngan) [inline test/confirmTransaction.js]: Can you fix the formatting on this?

```
} = await $util.buildTransaction(buyer, seller, {
  ...
})
```

**Review by ngan — CHANGES_REQUESTED** [review]

File: test/confirmTransaction.js [hunk 2]
```diff
@@ -2,48 +2,352 @@ const $util = require("./util")
...
+      it("passes the transaction's amount to the mediator", async () => {
+        let {
+          protocol,
+          transaction,
+          mediator
+        } = await $util.buildTransaction(
+          buyer, seller, { finalState: $util.states.Accepted }
+        )
+
+        let tx = await protocol.confirmTransaction(transaction.id, { from: buyer })
```
Comment (ngan) [inline test/confirmTransaction.js]: Do you need reference to `tx`?

**Comment by kraytruthg** [comment]
@ngan this is good to review

**Pull request merged**